ToDo/TodoApp/TodoApp/PageModels/CategoryDetailsPageModel.cs

```diff
@@ -68,8 +68,8 @@ public override void Init(object initData)
 
             if (initData is Category category)
             {
-                Category = category;
-                Device.BeginInvokeOnMainThread(() => this.Tasks = category.Items);
+                this.Category = this.TodoService.GetCategories().SingleOrDefault(c => c.ID == category.ID);
```

Comment:
Probably we should just pass just the ID here, not a whole fake object, which is not present in the collection. That would have prevented such a bug in the first place.

Reply:
I have changed it, now we pass only the ID